app/api/files/specs/storage.spec.ts

```diff
@@ -317,6 +318,126 @@ describe('storage', () => {
     });
   });
 
+  describe('listFiles', () => {
+    describe('when s3 flag is active', () => {
+      it('should list all files on disk', async () => {
+        testingTenants.changeCurrentTenant({
+          name: 'tenant1',
+          dbName: 'uwazi_development',
+          indexName: 'index',
+          featureFlags: {
+            s3Storage: true,
+          },
+        });
+        await storage.storeFile(
+          'file_created1.txt',
+          createReadStream(uploadsPath('documento.txt')),
+          'document'
+        );
+        await storage.storeFile(
+          'file_created2.txt',
+          createReadStream(uploadsPath('documento.txt')),
+          'document'
+        );
+        expect(await storage.listFiles()).toEqual([
+          { filename: 'already_uploaded.txt', type: 'uploads' },
+          { filename: 'file_created1.txt', type: 'uploads' },
+          { filename: 'file_created2.txt', type: 'uploads' },
+          { filename: 'file_to_be_deleted.txt', type: 'uploads' },
+          { filename: 'file_created.txt', type: 'segmentation' },
+        ]);
+      });
+    });
+
+    describe('when s3 flag is not active', () => {
+      it('should list all files on disk', async () => {
+        testingTenants.changeCurrentTenant({
+          name: 'tenant1',
+          dbName: 'uwazi_development',
+          indexName: 'index',
+          featureFlags: {
+            s3Storage: false,
+          },
+        });
+        await storage.storeFile(
+          'file_created.txt',
```

Comment:
I added new files to test the different file types.
The problem of the extra files is a tricky one. It's right now failing in the pipeline runs likely because of something around that. I will try to find a strategy to work that around.
Regarding clean up, I didn't do it because other test cases neither do it. But may be the solution is to clean up all of them.